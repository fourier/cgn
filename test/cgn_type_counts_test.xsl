<?xml version="1.0" encoding="utf-8"?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="2.0"
                xmlns:cgn="https://github.com/fourier/cgn"
                xmlns:xs="http://www.w3.org/2001/XMLSchema">

  <xsl:include href="../objects/cgn_includes.xsl"/>


  <xsl:template match="cgn:object" mode="type-counts">
    <xsl:copy>
      <!-- copy all attributes and fields -->
      <xsl:copy-of select="@* | node()"/>

      <!-- create a child type-counts -->
      <xsl:element name="type-counts">
        <xsl:call-template name="cgn:create-type-counts-xml"/>
      </xsl:element>

    </xsl:copy>
  </xsl:template>

  <xsl:template match="@*|node()" mode="type-counts">
    <xsl:copy>
      <xsl:apply-templates select="@* | node()" mode="type-counts"/>
    </xsl:copy>
  </xsl:template>

  
  <xsl:template name="create-type-counts-xml-test">
    <xsl:variable name="type-counts">
      <!-- construct cgn:object/object-counts  -->
      <xsl:apply-templates select="$cgn:preprocessed-objects" mode="type-counts" />
    </xsl:variable>

    <xsl:message>Testing cgn:create-type-counts-xml template</xsl:message>
    <!-- template for command line output -->
    <xsl:sequence select="$type-counts"/>

  </xsl:template>


</xsl:stylesheet>

