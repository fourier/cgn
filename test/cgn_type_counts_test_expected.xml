<?xml version="1.0" encoding="UTF-8"?>
<!-- default namespace for elements: cgn -->
<root xmlns="https://github.com/fourier/cgn"
      xmlns:cgn="https://github.com/fourier/cgn"
      xmlns:jcgn="https://github.com/fourier/cgn/java"><!-- this test verifies the cgn:create-type-counts-xml template --><!-- test if nothing happens with the normal field types,
       like primitive types or arrays of primitive types
  -->
   <objects cgn:package="com.test.phase5.field.types.normal"
            cgn:copyright=""
            cgn:author=""
            cgn:read-only="false"
            cgn:json="false"
            cgn:is-set="false">
      <object cgn:name="Test1"
              cgn:copyright=""
              cgn:author=""
              cgn:package="com.test.phase5.field.types.normal"
              cgn:read-only="false"
              cgn:json="false"
              cgn:is-set="false">
         <field cgn:name="field1" cgn:type="int"/>
         <field cgn:name="field2" cgn:type="double"/>
         <field cgn:name="field3" cgn:type="[int]"/>
         <field cgn:name="field4" cgn:type="[string]"/>
         <field cgn:name="field5" cgn:type="string"/>
         <field cgn:name="field6" cgn:type="date"/>
         <type-counts xmlns=""/>
      </object>
   </objects>
   <!-- test field types which are not arrays -->
   <objects cgn:package="com.test.type.counts"
            cgn:copyright=""
            cgn:author=""
            cgn:read-only="false"
            cgn:json="false"
            cgn:is-set="false"><!-- for this test counts should be:
         com.test.type.counts.helpers.one.SomeObject1 count=3
         com.test.type.counts.helpers.two.SomeObject1 count=3
         com.test.type.counts.helpers.three.SomeObject1 count=3
         com.test.type.counts.helpers.one.SomeObject2 count=2
         com.test.type.counts.helpers.two.SomeObject2 count=2
         com.test.type.counts.helpers.one.TheObject count=1
    -->
      <object cgn:name="Test1"
              cgn:copyright=""
              cgn:author=""
              cgn:package="com.test.type.counts"
              cgn:read-only="false"
              cgn:json="false"
              cgn:is-set="false">
         <field cgn:name="someobject1"
                cgn:type="com.test.type.counts.helpers.one.SomeObject1"/>
         <field cgn:name="someobjects1"
                cgn:type="[com.test.type.counts.helpers.two.SomeObject1]"/>
         <field cgn:name="someobjects-another"
                cgn:type="[com.test.type.counts.helpers.three.SomeObject1]"/>
         <field cgn:name="someobjects2"
                cgn:type="[com.test.type.counts.helpers.one.SomeObject2]"/>
         <field cgn:name="someobject2"
                cgn:type="com.test.type.counts.helpers.two.SomeObject2"/>
         <field cgn:name="theobject"
                cgn:type="com.test.type.counts.helpers.one.TheObject"/>
         <type-counts xmlns="">
            <fqdn type="com.test.type.counts.helpers.one.SomeObject1" count="3"/>
            <fqdn type="com.test.type.counts.helpers.two.SomeObject1" count="3"/>
            <fqdn type="com.test.type.counts.helpers.three.SomeObject1" count="3"/>
            <fqdn type="com.test.type.counts.helpers.one.SomeObject2" count="2"/>
            <fqdn type="com.test.type.counts.helpers.two.SomeObject2" count="2"/>
            <fqdn type="com.test.type.counts.helpers.one.TheObject" count="1"/>
         </type-counts>
      </object>
   </objects>
   <objects cgn:package="com.test.type.counts.helpers.one"
            cgn:copyright=""
            cgn:author=""
            cgn:read-only="false"
            cgn:json="false"
            cgn:is-set="false">
      <object cgn:name="SomeObject1"
              cgn:copyright=""
              cgn:author=""
              cgn:package="com.test.type.counts.helpers.one"
              cgn:read-only="false"
              cgn:json="false"
              cgn:is-set="false">
         <type-counts xmlns=""/>
      </object>
      <object cgn:name="SomeObject2"
              cgn:copyright=""
              cgn:author=""
              cgn:package="com.test.type.counts.helpers.one"
              cgn:read-only="false"
              cgn:json="false"
              cgn:is-set="false">
         <type-counts xmlns=""/>
      </object>
      <object cgn:name="TheObject"
              cgn:copyright=""
              cgn:author=""
              cgn:package="com.test.type.counts.helpers.one"
              cgn:read-only="false"
              cgn:json="false"
              cgn:is-set="false">
         <type-counts xmlns=""/>
      </object>
   </objects>
   <objects cgn:package="com.test.type.counts.helpers.two"
            cgn:copyright=""
            cgn:author=""
            cgn:read-only="false"
            cgn:json="false"
            cgn:is-set="false">
      <object cgn:name="SomeObject1"
              cgn:copyright=""
              cgn:author=""
              cgn:package="com.test.type.counts.helpers.two"
              cgn:read-only="false"
              cgn:json="false"
              cgn:is-set="false">
         <type-counts xmlns=""/>
      </object>
      <object cgn:name="SomeObject2"
              cgn:copyright=""
              cgn:author=""
              cgn:package="com.test.type.counts.helpers.two"
              cgn:read-only="false"
              cgn:json="false"
              cgn:is-set="false">
         <type-counts xmlns=""/>
      </object>
      <object cgn:name="TheObject"
              cgn:copyright=""
              cgn:author=""
              cgn:package="com.test.type.counts.helpers.two"
              cgn:read-only="false"
              cgn:json="false"
              cgn:is-set="false">
         <type-counts xmlns=""/>
      </object>
   </objects>
   <objects cgn:package="com.test.type.counts.helpers.three"
            cgn:copyright=""
            cgn:author=""
            cgn:read-only="false"
            cgn:json="false"
            cgn:is-set="false">
      <object cgn:name="SomeObject1"
              cgn:copyright=""
              cgn:author=""
              cgn:package="com.test.type.counts.helpers.three"
              cgn:read-only="false"
              cgn:json="false"
              cgn:is-set="false">
         <type-counts xmlns=""/>
      </object>
   </objects>
</root>
